--- a/target/linux/ar71xx/base-files/lib/ar71xx.sh
+++ b/target/linux/ar71xx/base-files/lib/ar71xx.sh
@@ -461,9 +461,9 @@ ar71xx_board_detect() {
 		name="gl-inet"
 		gl_inet_board_detect
 		;;
-	*"GL-AR150")
-        name="gl-ar150"
-        ;;
+	*"WiFi Pineapple NANO")
+        	name="gl-ar150"
+        	;;
 	*"EnGenius EPG5000")
 		name="epg5000"
 		;;
--- a/target/linux/ar71xx/files/arch/mips/ath79/mach-gl-ar150.c
+++ b/target/linux/ar71xx/files/arch/mips/ath79/mach-gl-ar150.c
@@ -125,4 +125,4 @@ static void __init gl_ar150_setup(void)
 	ath79_register_wmac(art + GL_AR150_CALDATA_OFFSET, art + GL_AR150_WMAC_MAC_OFFSET);
 }
 
-MIPS_MACHINE(ATH79_MACH_GL_AR150, "GL-AR150", "GL-AR150",gl_ar150_setup);
+MIPS_MACHINE(ATH79_MACH_GL_AR150, "GL-AR150", "WiFi Pineapple NANO",gl_ar150_setup);
